### PR TITLE
create group admins

### PR DESCRIPTION
**Pull request by Andrii-Kozak**
Знімок екрану з 2021-01-28 16-12-34

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FactoryBot.create_list(:user, 15)
 FactoryBot.create_list(:group, 10)
+FactoryBot.create_list(:user_group, 10)
```
Comment (pereslop) [inline db/seeds.rb:3]: You created 10 usergroups, but they are not associated with users, and  groups

File: spec/factories/user_groups.rb [hunk 2]
```diff
@@ -1,8 +1,17 @@
+FactoryBot.define do
+  factory :user_group do
+    group
+    user
+    admin { false }
+  end
+end
+
```
Comment (pereslop) [inline spec/factories/user_groups.rb:8]: Also add here a trait with admin: true

File: spec/models/user_group_spec.rb [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
+
   describe 'associations' do
     context 'when belong_to' do
       it 'user' do is_expected.to belong_to(:user) end
       it 'group' do is_expected.to belong_to(:group) end
     end
   end
+
+  it 'returns empty collection' do
+    expect(UserGroup.for_user(user)).to be_empty
```
Comment (pereslop) [inline spec/models/user_group_spec.rb:14]: Described_class

File: spec/models/user_group_spec.rb [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
+  let!(:user) { create(:user) }
+
   describe 'associations' do
     context 'when belong_to' do
       it 'user' do is_expected.to belong_to(:user) end
       it 'group' do is_expected.to belong_to(:group) end
     end
   end
+
+  it 'returns empty collection' do
```
Comment (pereslop) [inline spec/models/user_group_spec.rb:13]: Need one more test, to check if not empty

**Review by pereslop — APPROVED** [review]

File: spec/models/user_group_spec.rb [hunk 5]
```diff
@@ -1,19 +1,35 @@
...
   end
+
+  describe 'user_group moderators' do
+    it 'returns empty collection' do
+      expect(described_class.for_user(user)).to be_empty
+    end
+
+    it 'returns not empty collection' do
+      user_group.reload
+      expect(described_class.for_user(user)).not_to be_empty
```
Comment (pereslop) [inline spec/models/user_group_spec.rb:22]: also you can check if user is in collection
`      expect(described_class.for_user(user)).to include(user)`

File: spec/models/user_group_spec.rb [hunk 6]
```diff
@@ -1,19 +1,35 @@
...
     end
   end
+
+  describe 'user_group moderators' do
+    it 'returns empty collection' do
+      expect(described_class.for_user(user)).to be_empty
+    end
+
+    it 'returns not empty collection' do
+      user_group.reload
```
Comment (pereslop) [inline spec/models/user_group_spec.rb:21]: do you really need to reload user_group?

File: app/models/group.rb [hunk 7]
```diff
@@ -8,14 +8,18 @@ class Group < ApplicationRecord
...
+  has_many :users, through: :user_groups
 
   validates :group_name,  presence: true, length: { maximum: 50 }
   validates :description, presence: true, length: { maximum: 120 }
   validates :group_type, presence: true, inclusion: { in: GROUP_TYPE }
 
   mount_uploader :image, ImageUploader
   paginates_per 6
+
+  def user_moderator?(user)
```
Comment (pereslop) [inline app/models/group.rb:20]: this test needs specs)

File: app/controllers/groups_controller.rb [hunk 8]
```diff
@@ -25,13 +31,18 @@ def create
     end
   end
 
-  def update
-    if @group.update(group_params)
-      flash[:success] = "group \"#{@group.group_name}\"  updated"
-      redirect_to @group
+  def update # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    if @group.user_moderator?(current_user) || current_user.admin?
```
Comment (pereslop) [inline app/controllers/groups_controller.rb:35]: it is OK for not but in future you need to add policies to manage access

File: app/models/user_group.rb [hunk 9]
```diff
@@ -1,13 +1,17 @@
 class UserGroup < ApplicationRecord
   belongs_to :user
   belongs_to :group
+
+  scope :moderators, -> { where(moderator: true) }
```
Comment (pereslop) [inline app/models/user_group.rb:5]: this scope needs specs to)

File: app/controllers/groups_controller.rb [hunk 10]
```diff
@@ -12,11 +12,17 @@ def new
     @group = Group.new
   end
 
-  def edit; end
+  def edit
+    unless @group.user_moderator?(current_user) || current_user.admin?
```
Comment (pereslop) [inline app/controllers/groups_controller.rb:16]: it is OK for not but in future you need to add policies to manage access

**Review by pereslop — COMMENTED** [review]
one more: you need to check access on destroy action

File: spec/models/user_group_spec.rb [hunk 11]
```diff
@@ -1,19 +1,47 @@
 require 'rails_helper'
 
 RSpec.describe UserGroup, type: :model do
+  let!(:user) { create(:user) }
+  let!(:group) { create(:group) }
+  let!(:user_group) { create(:user_group, user: user, group: group, moderator: true) }
```
Comment (pereslop) [inline spec/models/user_group_spec.rb:6]: you dont need here `moderator: true`, you can use trait defined in UserGroup factory

File: spec/models/group_spec.rb [hunk 12]
```diff
@@ -29,6 +29,18 @@
       it 'users' do is_expected.to respond_to(:users) end
     end
   end
+
+  describe 'when user' do
+    let(:user_group) { create(:user_group, moderator: true) }
```
Comment (pereslop) [inline spec/models/group_spec.rb:34]: use trait 
`create(:user_group, :moderator)`

**Review by pereslop — APPROVED** [review]
